RNASEQ_SCHEMA = [
  {
    "name": "parent_sample_id",
    "description": "Unique HuBMAP or SenNet identifier of the sample (i.e., block, section or suspension) used to perform this assay. For example, for a RNAseq assay, the parent would be the suspension, whereas, for one of the imaging assays, the parent would be the tissue section. If an assay comes from multiple parent samples then this should be a comma separated list. Example: HBM386.ZGKG.235, HBM672.MKPK.442 or SNT232.UBHJ.322, SNT329.ALSK.102",
    "type": "text",
    "required": True,
    "regex": "^(?:HBM|SNT)\\d{3}\\.[A-Z]{4}\\.\\d{3}(?:,\\s*(?:HBM|SNT)\\d{3}\\.[A-Z]{4}\\.\\d{3})*$",
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "lab_id",
    "description": "An internal field labs can use it to add whatever ID(s) they want or need for dataset validation and tracking. This could be a single ID (e.g., \"Visium_9OLC_A4_S1\") or a delimited list of IDs (e.g., “9OL; 9OLC.A2; Visium_9OLC_A4_S1”). This field will not be accessible to anyone outside of the consortium and no effort will be made to check if IDs provided by one data provider are also used by another.",
    "type": "text",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "preparation_protocol_doi",
    "description": "DOI for the protocols.io page that describes the assay or sample procurement and preparation. For example for an imaging assay, the protocol might begin with staining of a section and finalize with the creation of an OME-TIFF file. In this case the protocol would include any image processing steps required to create the OME-TIFF file. Example: https://dx.doi.org/10.17504/protocols.io.eq2lyno9qvx9/v1.",
    "type": "text",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "dataset_type",
    "description": "The specific type of dataset being produced.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": "RNAseq",
    "permissible_values": [
      "HiFi-Slide",
      "SNARE-seq2",
      "MIBI",
      "Visium (no probes)",
      "DESI",
      "Auto-fluorescence",
      "Confocal",
      "Xenium",
      "Stereo-seq",
      "Visium (with probes)",
      "Molecular Cartography",
      "CosMx",
      "DBiT-seq",
      "SIMS",
      "Cell DIVE",
      "CODEX",
      "GeoMx (NGS)",
      "MUSIC",
      "CyCIF",
      "Light Sheet",
      "MALDI",
      "seqFISH",
      "2D Imaging Mass Cytometry",
      "ATACseq",
      "Histology",
      "Enhanced Stimulated Raman Spectroscopy (SRS)",
      "DART-FISH",
      "Resolve",
      "RNAseq",
      "Singular Genomics G4X",
      "MERFISH",
      "LC-MS",
      "nanoSPLITS",
      "10X Multiome",
      "GeoMx (nCounter)",
      "RNAseq (with probes)",
      "PhenoCycler",
      "Second Harmonic Generation (SHG)",
      "Thick section Multiphoton MxIF",
      "MS Lipidomics",
      "CyTOF"
    ]
  },
  {
    "name": "analyte_class",
    "description": "Analytes are the target molecules being measured with the assay.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "DNA + RNA",
      "Nucleic acid + protein",
      "Chromatin",
      "RNA",
      "Metabolite",
      "Unsaturated lipid",
      "Lipid + metabolite",
      "Saturated lipid",
      "Lipid",
      "Protein",
      "Fluorochrome",
      "Collagen",
      "DNA",
      "Peptide",
      "Polysaccharide",
      "Endogenous fluorophore"
    ]
  },
  {
    "name": "is_targeted",
    "description": "Specifies whether or not a specific molecule(s) is/are targeted for detection/measurement by the assay (\"Yes\" or \"No\"). The CODEX analyte is protein.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Yes",
      "No"
    ]
  },
  {
    "name": "acquisition_instrument_vendor",
    "description": "An acquisition instrument is the device that contains the signal detection hardware and signal processing software. Assays generate signals such as light of various intensities or color or signals representing the molecular mass.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "BGI Genomics",
      "Cytiva",
      "Thermo Fisher Scientific",
      "Zeiss Microscopy",
      "GE Healthcare",
      "Leica Microsystems",
      "Akoya Biosciences",
      "NanoString",
      "Element Biosciences",
      "Andor",
      "Huron Digital Pathology",
      "Illumina",
      "Ionpath",
      "In-House",
      "Resolve Biosciences",
      "Singular Genomics",
      "Vizgen",
      "Standard BioTools (Fluidigm)",
      "Sciex",
      "Bruker",
      "Evident Scientific (Olympus)",
      "Keyence",
      "Leica Biosystems",
      "10x Genomics",
      "Hamamatsu",
      "Motic"
    ]
  },
  {
    "name": "acquisition_instrument_model",
    "description": "Manufacturers of an acquisition instrument may offer various versions (models) of that instrument with different features or sensitivities. Differences in features or sensitivities may be relevant to processing or interpretation of the data.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "SCN400",
      "STELLARIS 5",
      "BZ-X710",
      "Not applicable",
      "MoticEasyScan One",
      "EVOS M7000",
      "NovaSeq X",
      "NanoZoomer 2.0-HT",
      "timsTOF Ultra 2",
      "Lightsheet 7",
      "Phenocycler-Fusion 1.0",
      "DNBSEQ-T7",
      "timsTOF Pro",
      "Unknown",
      "AVITI",
      "timsTOF Pro 2",
      "Q Exactive UHMR",
      "Q Exactive",
      "timsTOF SCP",
      "Zyla 4.2 sCMOS",
      "Helios",
      "Custom: Multiphoton",
      "QTRAP 5500",
      "timsTOF Ultra",
      "BZ-X800",
      "CyTOF 2",
      "G4X Spatial Sequencer",
      "NextSeq 500",
      "NanoZoomer S360",
      "Hyperion Imaging System",
      "NovaSeq X Plus",
      "CyTOF XT",
      "NanoZoomer-SQ",
      "NextSeq 550",
      "Digital Spatial Profiler",
      "timsTOF FleX",
      "timsTOF FleX MALDI-2",
      "NanoZoomer S210",
      "BZ-X810",
      "Axio Observer 7",
      "IN Cell Analyzer 2200",
      "timsTOF HT",
      "PhenoImager Fusion",
      "DM6 B",
      "Phenocycler-Fusion 2.0",
      "Aperio CS2",
      "Orbitrap Fusion Lumos Tribrid",
      "Resolve Biosciences Molecular Cartography",
      "MALDI timsTOF Flex Prototype",
      "TissueScope LE Slide Scanner",
      "VS200 Slide Scanner",
      "Axio Observer 5",
      "Axio Observer 3",
      "HiSeq 2500",
      "Orbitrap Eclipse Tribrid",
      "MERSCOPE",
      "NextSeq 2000",
      "NovaSeq 6000",
      "In-House",
      "HiSeq 4000",
      "Aperio AT2",
      "MIBIscope",
      "NanoZoomer S60",
      "CosMx Spatial Molecular Imager",
      "MERSCOPE Ultra",
      "Axio Scan.Z1",
      "Q Exactive HF",
      "Xenium Analyzer"
    ]
  },
  {
    "name": "source_storage_duration_value",
    "description": "How long was the source material (parent) stored, prior to this sample being processed.",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "source_storage_duration_unit",
    "description": "The time duration unit of measurement",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "hour",
      "month",
      "year",
      "day",
      "minute"
    ]
  },
  {
    "name": "time_since_acquisition_instrument_calibration_value",
    "description": "The amount of time since the acquisition instrument was last serviced or calibrated. This provides a metric for assessing drift in data capture.",
    "type": "number",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "time_since_acquisition_instrument_calibration_unit",
    "description": "The time unit of measurement",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "month",
      "year",
      "day"
    ]
  },
  {
    "name": "contributors_path",
    "description": "The path to the file with the ORCID IDs for all contributors of this dataset (e.g., \"./extras/contributors.tsv\" or \"./contributors.tsv\"). This is an internal metadata field that is just used for ingest.",
    "type": "text",
    "required": True,
    "regex": "^(?:\\.\\/.*|\\w.*)\\.tsv$",
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "data_path",
    "description": "The top level directory containing the raw and/or processed data. For a single dataset upload this might be \".\" where as for a data upload containing multiple datasets, this would be the directory name for the respective dataset. For instance, if the data is within a directory called \"TEST001-RK\" use syntax \"./TEST001-RK\" for this field. If there are multiple directory levels, use the format \"./TEST001-RK/Run1/Pass2\" in which \"Pass2\" is the subdirectory where the single dataset's data is stored. This is an internal metadata field that is just used for ingest.",
    "type": "text",
    "required": True,
    "regex": "^(?:\\.|\\.\\/.+|\\w.*)$",
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "barcode_offset",
    "description": "Positions in the read at which the cell or capture spot barcodes start. Cell and capture spot barcodes are, for example, 3 x 8 bp sequences that are spaced by constant sequences (the offsets). First barcode at position 0, then 38, then 76. This should be included when constructing sequencing libraries with a non-commercial kit.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      0,
      8,
      20,
      "1,27",
      "0,38,76",
      "10,48,78",
      "10,48,86",
      "Not applicable"
    ]
  },
  {
    "name": "barcode_read",
    "description": "Which read file contains the cell or capture spot barcode. This should be included when constructing sequencing libraries with a non-commercial kit. This field is required if the source material is barcoded. This field is used to determine which analysis pipeline to run.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Read 1 (R1)",
      "Not applicable",
      "Read 2 (R2)"
    ]
  },
  {
    "name": "barcode_size",
    "description": "Length of the cell or capture spot barcode in base pairs. Cell and capture spot barcodes are, for example, 3 x 8 bp sequences that are spaced by constant sequences, the offsets. This should be included when constructing sequencing libraries with a non-commercial kit. This field is required if the source material is barcoded.  This field is used to determine which analysis pipeline to run.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      14,
      16,
      40,
      "8,8,8",
      "8,6",
      "Not applicable"
    ]
  },
  {
    "name": "umi_offset",
    "description": "Position in the read at which the UMI barcode starts. This should be included when constructing sequencing libraries with a non-commercial kit.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      0,
      16,
      36,
      "Not applicable"
    ]
  },
  {
    "name": "umi_read",
    "description": "Which read file contains the UMI barcode. This should be included when constructing sequencing libraries with a non-commercial kit.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Read 1 (R1)",
      "Not applicable",
      "Read 2 (R2)"
    ]
  },
  {
    "name": "umi_size",
    "description": "Length of the umi barcode in base pairs. This should be included when constructing sequencing libraries with a non-commercial kit. This field is required if UMI are present. This field is used to determine which analysis pipeline to run.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      8,
      9,
      10,
      12,
      "Not applicable"
    ]
  },
  {
    "name": "assay_input_entity",
    "description": "This is the entity from which the analyte is being captured. For example, for bulk sequencing this would be \"tissue\", while it would be \"single cell\" for single cell sequencing. This field is used to determine which analysis pipeline to run.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "single cell",
      "tissue (bulk)",
      "spot",
      "single nucleus",
      "area of interest"
    ]
  },
  {
    "name": "number_of_input_cells_or_nuclei",
    "description": "How many cells or nuclei were input to the assay? This is typically not available for preparations working with bulk tissue.",
    "type": "number",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "amount_of_input_analyte_value",
    "description": "The amount of RNA or DNA input to the assay, typically measured by a Qubit, BioAnalyzer, or TapeStation. In most single cell/nuclei assays, this value isn't available.",
    "type": "number",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "amount_of_input_analyte_unit",
    "description": "Units of amount of entity input to assay value",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "ug",
      "ng"
    ]
  },
  {
    "name": "library_adapter_sequence",
    "description": "5’ and/or 3’ read adapter sequences used as part of the library preparation protocol to render the library compatible with the sequencing protocol and instrumentation. This should be provided as comma-separated list of key:value pairs (adapter name:sequence).",
    "type": "text",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_average_fragment_size",
    "description": "Average size of sequencing library fragments estimated via gel electrophoresis or bioanalyzer/tapestation. Numeric value in base pairs (bp).",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_input_amount_value",
    "description": "The amount of cDNA, after amplification, that was used for library construction.",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_input_amount_unit",
    "description": "unit of library input amount value",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "ul",
      "ng"
    ]
  },
  {
    "name": "library_output_amount_value",
    "description": "Total amount (eg. nanograms) of library after the clean-up step of final pcr amplification step. Answer the question: What is the Qubit measured concentration (ng/ul) times the elution volume (ul) after the final clean-up step?",
    "type": "number",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_output_amount_unit",
    "description": "Units of library final yield.",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "ul",
      "ng"
    ]
  },
  {
    "name": "library_concentration_value",
    "description": "The concentration value of the pooled library samples submitted for sequencing.",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_concentration_unit",
    "description": "Unit of library concentration value.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "ng/ul",
      "nM"
    ]
  },
  {
    "name": "library_layout",
    "description": "Whether the library was generated for single-end or paired end sequencing",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "single-end",
      "paired-end"
    ]
  },
  {
    "name": "number_of_iterations_of_cdna_amplification",
    "description": "This is the amplification of the cDNA prior to library construction. This is typically a PCR amplification, while for linear amplification methods like aRNA this would be the number of rounds of aRNA.",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "number_of_pcr_cycles_for_indexing",
    "description": "Number of PCR cycles performed in order to add adapters and amplify the library. This does not include the cDNA amplification which is captured in the \"number of iterations of cDNA amplification\" field.",
    "type": "number",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "library_preparation_kit",
    "description": "Reagent kit used for library preparation",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "10X Genomics; Chromium Next GEM Single Cell 3' HT Kit v3.1, 8 rxns; PN 1000370",
      "10X Genomics; Chromium Next GEM Automated Single Cell 5' Kit v2, 4 rxns; PN 1000298",
      "10X Genomics; Visium Spatial Gene Expression Slide and Reagent Kit, 4 slides, 16 reactions; PN 1000184",
      "Parse Biosciences; Evercode WT v2 Kit, 48 rxns; PN ECW02030",
      "10X Genomics; Chromium Next GEM Single Cell 5' Kit v2, 16 rxns; PN 1000263",
      "10X Genomics; Visium Spatial for FFPE Gene Expression Kit, Human Transcriptome, 1 slides, 4 reactions; PN 1000338",
      "10X Genomics; Chromium Next GEM Single Cell 3' Kit v3.1, 16 rxns; PN 1000268",
      "10x Genomics; Library Construction Kit C, 16 rxns; PN 1000694",
      "10x Genomics; Visium CytAssist Spatial Gene Expression for FFPE, Mouse Transcriptome, 6.5mm, 4 rxns; PN 1000521",
      "10X Genomics; Chromium Next GEM Single Cell 3' HT Kit v3.1, 48 rxns; PN 1000348",
      "10X Genomics; Chromium Next GEM Single Cell 3' GEM, Library & Gel Bead Kit v3.1, 16 rxns; PN 1000121",
      "10x Genomics; Library Construction Kit C, 4 rxns; PN 1000689",
      "10X Genomics; Visium CytAssist Spatial Gene Expression for FFPE, Human Transcriptome, 11 mm, 2 reactions; PN 1000522",
      "Illumina; TruSeq Stranded mRNA Library Prep (48 samples); PN 20020594",
      "10X Genomics; Automated Library Construction Kit, 24 rxns; PN 1000428",
      "New England BioLabs; NEBNext Ultra II RNA Library Prep Kit for Illumina; PN E7770",
      "10X Genomics; Chromium Next GEM Automated Single Cell 5' Kit v2, 24 rxns; PN 1000290",
      "10X Genomics; Visium Spatial Gene Expression Slide and Reagent Kit, 1 slides, 4 reactions; PN 1000187",
      "10x Genomics; Chromium GEM-X Single Cell 5' Kit v3, 16 rxns; PN 1000699",
      "Parse Biosciences; Evercode WT Mini v2 Kit, 12 rxns; PN ECW02010",
      "Illumina; TruSeq Stranded mRNA Library Prep (96 samples); PN 20020595",
      "10X Genomics; Chromium Next GEM Single Cell 5' Kit v2, 4 rxns; PN 1000265",
      "10X Genomics; Chromium NextGem Single Cell Multiome ATAC + Gene Expression Reagent Bundle, 4 rxn; PN 1000285",
      "10X Genomics; Chromium Next GEM Single Cell Fixed RNA Hybridization & Library Kit, 4 rxns; PN 1000415",
      "Custom",
      "10X Genomics; Chromium Single Cell 3' GEM, Library & Gel Bead Kit v3, 4 rxns; PN 1000092",
      "10X Genomics; Chromium Single Cell 3' Library & Gel Bead Kit, 4 rxns; PN 120267",
      "10X Genomics; Visium Spatial for FFPE Gene Expression Kit, Mouse Transcriptome, 4 rxns; PN 1000339",
      "10X Genomics; Chromium NextGem Single Cell Multiome ATAC + Gene Expression Reagent Bundle, 16 rxn; PN 1000283",
      "10X Genomics; Chromium Next GEM Single Cell 3' Kit v3.1, 4 rxns; PN 1000269",
      "10X Genomics; Visium CytAssist Spatial Gene Expression for FFPE, Human Transcriptome, 6.5mm, 4 reactions; PN 1000520"
    ]
  },
  {
    "name": "sample_indexing_kit",
    "description": "Indexes are needed for multiplexing sequencing libraries for simultaneous sequencing (pooling) and proper attachment to the Illumina flowcell. Each indexing kit would have a number of compatible sequences (\"sample indexing sets\") that are used to label some number of samples (the number of sets depend on the kit).",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Illumina; TruSeq RNA CD Index Plate (96 Indexes, 96 Samples); PN 20019792",
      "Not applicable",
      "Illumina; TruSeq RNA Single Indexes Set A (12 Indexes, 48 Samples); PN 20020492",
      "Illumina; TruSeq RNA Single Indexes Set B (12 Indexes, 48 Samples); PN 20020493",
      "10X Genomics; Chromium i7 Sample Index Plate (96 rxn); PN 220103",
      "NanoString Technologies; GeoMx Seq Code Pack; PN GMX-NGS-SEQ-AB",
      "NanoString Technologies; GeoMx Seq Code Pack; PN GMX-NGS-SEQ-CD",
      "NanoString Technologies; GeoMx Seq Code Pack; PN GMX-NGS-SEQ-EF",
      "NanoString Technologies; GeoMx Seq Code Pack; PN GMX-NGS-SEQ-GH",
      "Custom",
      "Parse Biosciences; Fragmentation Reagents; PN WX100",
      "Integrated DNA Technologies: Custom DNA Oligos",
      "Parse Biosciences; UDI Plate - WT; PN UDI1001",
      "10X Genomics; Dual Index Kit TS, Set A; PN 1000251",
      "Illumina; IDT for Illumina - TruSeq RNA UD Indexes v2 (96 Indexes, 96 Samples); PN 20040871",
      "10X Genomics; Single Index Kit N, Set A (96 rxn); PN 1000212",
      "10X Genomics; Dual Index Kit TT, Set A (96 rxn); PN 1000215"
    ]
  },
  {
    "name": "sample_indexing_set",
    "description": "The specific sequencing barcode index set used, selected from the sample indexing kit. Example: For 10X this might be \"SI-GA-A1\", for Nextera \"N505 - CTCCTTAC\"",
    "type": "text",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "is_technical_replicate",
    "description": "Is the sequencing reaction run in replicate, \"Yes\" or \"No\". If \"Yes\", FASTQ files in dataset need to be merged.",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Yes",
      "No"
    ]
  },
  {
    "name": "expected_entity_capture_count",
    "description": "Number of cells, nuclei or capture spots expected to be captured by the assay. For Visium this is the total number of spots covered by tissue, within the capture area.",
    "type": "number",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "sequencing_reagent_kit",
    "description": "Reagent kit used for sequencing",
    "type": "categorical",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "Illumina; NovaSeq X Series 1.5B Reagent Kit (300 Cycle); PN 20104705",
      "Illumina; NovaSeq 6000 S1 Reagent v1.5 Kit (200 Cycles); PN 20028318",
      "Illumina; NextSeq 2000 P3 Reagent Kit (300 Cycles); PN 20040561",
      "Illumina; NextSeq 1000/2000 P2 Reagent v3 Kit (200 Cycles); PN 20046812",
      "Illumina; NovaSeq X Series 1.5B Reagent Kit (200 Cycle); PN 20104704",
      "Illumina; NovaSeq X Series 10B Reagent Kit (300 Cycle); PN 20085594",
      "Element Biosciences; AVITI 2x75 Sequencing Kit Cloudbreak FS High Output (184 Cycles); PN 860-00015",
      "Illumina; NovaSeq X Series 1.5B Reagent Kit (100 Cycle); PN 20104703",
      "Illumina; NovaSeq X Series 25B Reagent Kit (300 Cycle); PN 20104706",
      "Illumina; NovaSeq 6000 S1 Reagent v1.5 Kit (100 Cycles); PN 20028319",
      "Illumina; NovaSeq 6000 SP Reagent v1.5 Kit (100 Cycles); PN 20028401",
      "Illumina; HiSeq 3000/4000 PE Cluster Kit PE-410-1001; PN 1000283",
      "Illumina; NextSeq 1000/2000 P2 Reagent v3 Kit (300 Cycles); PN 20046813",
      "Illumina; NextSeq 2000 P3 Reagents Kit (100 Cycles); PN 20040559",
      "Illumina; NovaSeq 6000 S1 Reagent v1.5 Kit (300 Cycles); PN 20028317",
      "Custom",
      "Illumina; NextSeq 1000/2000 P2 Reagent v3 Kit (100 Cycles); PN 20046811",
      "Illumina; NovaSeq X Series 10B Reagent Kit (200 Cycle); PN 20085595",
      "Illumina; NextSeq 500/550 Hi Output Kit 150 Cycles; v2.5; PN 20024907",
      "Illumina; NextSeq 500/550 Mid Output Kit 150 Cycles v2.5; PN 20024904",
      "Illumina; NovaSeq 6000 S1 Reagent Kit (200 Cycles); PN 20012864",
      "Illumina; NovaSeq 6000 S4 Reagent v1.5 Kit (200 Cycles); PN 20028313",
      "Illumina; NextSeq 500/550 Hi Output Kit 75 Cycles v2.5; PN 20024906",
      "Illumina; NovaSeq 6000 S4 Reagent Kit v1.5 (300 cycles); PN 20028312",
      "Illumina; NovaSeq X Series 10B Reagent Kit (100 Cycle); PN 20085596",
      "Illumina; NovaSeq 6000 S2 Reagent v1.5 Kit (100 Cycles); PN 20028316"
    ]
  },
  {
    "name": "sequencing_read_format",
    "description": "Number of sequencing cycles in each round of sequencing (i.e., Read1, i7 index, i5 index, and Read2). This is reported as a comma-delimited list. Example: For 10X snATAC-seq (R1,Index,R2,R3) this might be: 50,8,16,50. For SNARE-seq2 this might be: 75,94,8,75",
    "type": "text",
    "required": True,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "sequencing_batch_id",
    "description": "The ID for the sequencing run. This could, for example, be the chip ID and should allow users the ability to determine which samples were processed together in a sequencing run. It is recommended that data providers prefix the ID with the center name, to prevent values overlapping across centers.",
    "type": "text",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "capture_batch_id",
    "description": "A lab-generated ID to identify which cells were captured at the same time. This would, for example, be an ID to denote which datasets were derived from a single 10X Genomics Chromium Controller run. In the case of the 10X Controller this could be the chip ID and would allow users the ability to determine which samples were processed together in a Chromium controller. It is recommended that data providers prefix the ID with the center name, to prevent values overlapping across centers.",
    "type": "text",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": None
  },
  {
    "name": "preparation_instrument_vendor",
    "description": "The manufacturer of the instrument used to prepare (staining/processing) the sample for the assay. If an automatic slide staining method was indicated this field should list the manufacturer of the instrument.",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "In-House",
      "Leica Biosystems",
      "Not applicable",
      "Thermo Fisher Scientific",
      "Roche Diagnostics",
      "HTX Technologies",
      "10x Genomics",
      "Hamamatsu",
      "SunChrom"
    ]
  },
  {
    "name": "preparation_instrument_model",
    "description": "Manufacturers of a staining system instrument may offer various versions (models) of that instrument with different features. Differences in features or sensitivities may be relevant to processing or interpretation of the data.",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "NanoZoomer S210",
      "Not applicable",
      "Sublimator",
      "EVOS M7000",
      "Chromium Controller",
      "NanoZoomer S360",
      "NanoZoomer S60",
      "Chromium X",
      "AutoStainer XL",
      "Visium CytAssist",
      "SunCollect Sprayer",
      "M3+ Sprayer",
      "Discovery Ultra",
      "ST5020 Multistainer",
      "Chromium iX",
      "Chromium Connect",
      "M5 Sprayer",
      "TM-Sprayer"
    ]
  },
  {
    "name": "preparation_instrument_kit",
    "description": "The reagent kit used with the preparation instrument.",
    "type": "categorical",
    "required": False,
    "regex": None,
    "default_value": None,
    "permissible_values": [
      "10X Genomics; Chromium Next GEM Chip G Single Cell Kit, 16 rxns; PN 1000127",
      "10X Genomics; Chromium Next GEM Chip K Automated Single Cell Kit, 48 rxns; PN 1000289",
      "10X Genomics; Chromium NextGem Single Cell Multiome ATAC + Gene Expression Reagent Bundle, 16 rxn; PN 1000283",
      "10X Genomics; Chromium NextGem Single Cell Multiome ATAC + Gene Expression Reagent Bundle, 4 rxn; PN 1000285",
      "10x Genomics; Chromium GEM-X Single Cell 3' Kit v4, 16 rxns; PN 1000691",
      "10X Genomics; Chromium Next GEM Chip K Single Cell Kit, 48 rxns; PN 1000286",
      "10X Genomics; Chromium Next GEM Chip G Single Cell Kit, 48 rxns; PN 1000120",
      "10X Genomics; Visium FFPE Reagent Kit v2-Small, PN 1000436",
      "Custom",
      "10X Genomics; Chromium Next GEM Chip Q Single Cell Kit, 16 rxns; PN 1000422",
      "10X Genomics; Chromium Next GEM Chip K Single Cell Kit, 16 rxns; PN 1000287"
    ]
  },
  {
    "name": "metadata_schema_id",
    "description": "The string that serves as the definitive identifier for the metadata schema version and is readily interpretable by computers for data validation and processing. Example: 22bc762a-5020-419d-b170-24253ed9e8d9",
    "type": "text",
    "required": True,
    "regex": None,
    "default_value": "944e5fa0-f68b-4bdd-8664-74a3909429a9",
    "permissible_values": None
  }
]
